### PR TITLE
FAI-17961: Add AI commit metrics support for Cursor connector

### PR DESCRIPTION
**Pull request by matiaslcoulougian**
## Summary
- Add new AI commit metrics stream to cursor source with pagination support  
- Create destination converter mapping AI metrics to vcs_AssistantMetric models
- Track Tab, Composer, and Manual line contributions separately as Custom metrics. 

### Source (cursor-source)
- **New Stream**: `AiCommitMetrics` class extending `AirbyteStreamBase`
- **API Integration**: Added `getAiCommitMetrics()` method to Cursor client with pagination
- **Schema**: Created JSON schema for AI commit metrics data structure
- **Types**: Added `AiCommitMetricItem` type to `faros-airbyte-common`
- **Tests**: Comprehensive test coverage with mock API responses

### Destination (airbyte-faros-destination) 
- **New Converter**: `AiCommitMetrics` class extending `CursorConverter`
- **Metric Mapping**: Maps commit data to `vcs_AssistantMetric` records:
  - `tabLinesAdded` → Custom with feature="Tab"
  - `composerLinesAdded` → Custom with feature="Composer" 
  - `nonAiLinesAdded` → Custom with feature="Manual"
- **Repository Tracking**: Includes repository references when available
- **Feature Enum**: Consolidated all Feature enums into common module
- **Tests**: Updated snapshots with new converter output

## Test Coverage
✅ Source stream tests (API mocking, data transformation)  
✅ Destination converter tests (Faros model generation)
✅ All existing tests still passing
✅ Comprehensive edge case coverage (null values, missing repos, etc.)

## Technical Implementation
- Uses GET `/analytics/ai-code/commits` endpoint with query parameters
- Handles pagination automatically with page size limits
- Supports incremental sync with cursor-based state management  
- Maps commit timestamps for proper chronological ordering
- Extracts organization info from repository names when available

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/cursor/common.ts [hunk 1]
```diff
@@ -67,17 +78,15 @@ export abstract class CursorConverter extends Converter {
...
-          },
+          organization,
           user: {uid: userEmail, source: this.streamName.source},
           tool: {
             category: VCSToolCategory.CodingAssistant,
             detail: VCSToolDetail.Cursor,
           },
           ...(model && {model}),
           ...(feature && {feature}),
+          ...(repository && {repository}),
```
Comment (dbruno21) [inline destinations/airbyte-faros-destination/src/converters/cursor/common.ts:89]: Should we add to the uid generation? (same as model and feature)

File: destinations/airbyte-faros-destination/src/converters/cursor/ai_commit_metrics.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    ctx: StreamContext
+  ): Promise<ReadonlyArray<DestinationRecord>> {
+    const metric = record.record.data as AiCommitMetricItem;
+
+    if (!metric.userEmail || !metric.commitHash) {
+      return [];
+    }
+
+    const res: DestinationRecord[] = [];
+    const timestamp = metric.commitTs || metric.createdAt;
```
[the remark this answers is not in the document]
Comment (matiaslcoulougian) [reply destinations/airbyte-faros-destination/src/converters/cursor/ai_commit_metrics.ts:37]: image
🤷🏽🤷🏽

File: sources/cursor-source/src/cursor.ts [hunk 3]
```diff
@@ -134,6 +136,53 @@ export class Cursor {
...
+  ): AsyncGenerator<AiCommitMetricItem> {
+    let page = 1;
+    let hasNextPage = true;
+
+    while (hasNextPage) {
+      const params = new URLSearchParams({
+        startDate,
+        endDate,
+        page: page.toString(),
+        pageSize: DEFAULT_PAGE_SIZE.toString(),
```
[the remark this answers is not in the document]
Comment (matiaslcoulougian) [reply sources/cursor-source/src/cursor.ts:151]: Devin is working on it 😉
